packages/site/cms-site/src/iframebridge/IFrameBridge.tsx

```diff
@@ -149,6 +149,18 @@ export const IFrameBridgeProvider = ({ children }: PropsWithChildren) => {
         }
     }, [recalculatePreviewElementsData]);
 
+    useEffect(() => {
+        previewElements.forEach((previewElement) => {
+            previewElement.element.addEventListener(PREVIEW_ELEMENT_SCROLLED_INTO_VIEW_EVENT, recalculatePreviewElementsData);
+        });
+
+        return () => {
+            previewElements.forEach((previewElement) => {
+                previewElement.element.removeEventListener(PREVIEW_ELEMENT_SCROLLED_INTO_VIEW_EVENT, recalculatePreviewElementsData);
+            });
+        };
+    }, [previewElements, recalculatePreviewElementsData]);
```

Comment:
Have you considered to instead listen to the scroll event? We could debounce the handling, then we'd know when the scrolling stops.

Reply:
Yes, the problem is, we don't know which element actually scrolls and there's no easy way to find this out. 

I considered recursively checking each parent up to `<html />` to see if there's any element with `overflow: auto` or `overflow: scroll` but with `swiper` for example, there isn't even any such element, as the scrolling is handled in some other way.

Reply:
I thought about listen to scroll and recalculating all previews, like you did in https://github.com/vivid-planet/comet/pull/3096. Or will this drastically impact performance?

Reply:
> I thought about listen to scroll and recalculating all previews, like you did in #3096

We'd have to listen to the scroll of a specific html element and there's no easy method to find the correct one that I'm aware of.

And I'm not sure what you're referring to in #3096, I don't think this had anything to do with scrolling 🤔 

Or am I misunderstanding something?